Lib/glyphsLib/builder/sources.py

```diff
@@ -151,30 +151,21 @@ def _to_designspace_source_layer(self):
 
     # First, collect all brace layers in the font and which glyphs and which masters
     # they belong to.
-    layer_name_to_master_ids = collections.defaultdict(set)
-    layer_name_to_glyph_names = collections.defaultdict(list)
+    layer_to_master_ids = collections.defaultdict(set)
+    layer_to_glyph_names = collections.defaultdict(list)
     for glyph in self.font.glyphs:
         for layer in glyph.layers:
-            if (
-                "{" in layer.name
-                and "}" in layer.name
-                and ".background" not in layer.name
-            ):
-                layer_name_to_master_ids[layer.name].add(layer.associatedMasterId)
-                layer_name_to_glyph_names[layer.name].append(glyph.name)
+            if layer._is_brace_layer():
+                key = (layer.name, tuple(layer._brace_coordinates()))
+                layer_to_master_ids[key].add(layer.associatedMasterId)
+                layer_to_glyph_names[key].append(glyph.name)
 
     # Next, insert the brace layers in a defined location in the existing designspace.
     designspace = self._designspace
     layers_to_insert = collections.defaultdict(list)
-    for layer_name, master_ids in layer_name_to_master_ids.items():
-        # Construct coordinates first...
-        brace_coordinates = [
-            float(c)
-            for c in layer_name[
-                layer_name.index("{") + 1 : layer_name.index("}")
-            ].split(",")
-        ]
-
+    for key, master_ids in layer_to_master_ids.items():
```

Comment:
optional: you could unpack the `key` directly in the for loop variable as `for (layer_name, brace_coordinates), master_ids in ...`

Reply:
I need to convert `brace_coordinates` to a list anyway, so it wouldn’t save much lines.